pkg/vulnsrc/alma/alma.go

```diff
@@ -0,0 +1,169 @@
+
+	bolt "go.etcd.io/bbolt"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy-db/pkg/db"
+	"github.com/aquasecurity/trivy-db/pkg/types"
+	"github.com/aquasecurity/trivy-db/pkg/utils"
+	"github.com/aquasecurity/trivy-db/pkg/vulnsrc/vulnerability"
+)
+
+const (
+	almaDir = "alma"
+)
+
+var (
+	platformFormat = "alma %s"
+)
+
+type VulnSrc struct {
+	dbc db.Operation
+}
+
+func NewVulnSrc() VulnSrc {
+	return VulnSrc{
+		dbc: db.Config{},
+	}
+}
+
+func (vs VulnSrc) Name() string {
+	return vulnerability.Alma
+}
+
+func (vs VulnSrc) Update(dir string) error {
+	rootDir := filepath.Join(dir, "vuln-list", almaDir)
+	errata := map[string][]erratum{}
+	err := utils.FileWalk(rootDir, func(r io.Reader, path string) error {
+		var erratum erratum
+		if err := json.NewDecoder(r).Decode(&erratum); err != nil {
+			return xerrors.Errorf("failed to decode Alma erratum: %w", err)
+		}
+
+		dirs := strings.Split(path, string(filepath.Separator))
+		if len(dirs) < 3 {
+			log.Printf("invalid path: %s\n", path)
+			return nil
+		}
+
+		version := dirs[len(dirs)-3]
+		errata[version] = append(errata[version], erratum)
+		return nil
+	})
+	if err != nil {
+		return xerrors.Errorf("error in Alma walk: %w", err)
+	}
+
+	if err := vs.save(errata); err != nil {
+		return xerrors.Errorf("error in Alma save: %w", err)
+	}
+
+	return nil
+}
+
+func (vs VulnSrc) save(errataVer map[string][]erratum) error {
+	err := vs.dbc.BatchUpdate(func(tx *bolt.Tx) error {
+		for version, errata := range errataVer {
+			if err := vs.commit(tx, version, errata); err != nil {
+				return xerrors.Errorf("error in save Alma %s: %w", version, err)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return xerrors.Errorf("error in db batch update: %w", err)
+	}
+	return nil
+}
+
+func (vs VulnSrc) commit(tx *bolt.Tx, version string, errata []erratum) error {
+	platformName := fmt.Sprintf(platformFormat, version)
+	for _, erratum := range errata {
+		references := []string{}
+		for _, ref := range erratum.References {
+			if ref.Type != "cve" {
+				references = append(references, ref.Href)
+			}
+		}
+
+		for _, ref := range erratum.References {
+			if ref.Type != "cve" {
+				continue
+			}
+
+			cveID := ref.Title
+			for _, pkg := range erratum.Pkglist.Packages {
+				advisory := types.Advisory{
+					FixedVersion: constructVersion(pkg.Epoch, pkg.Version, pkg.Release),
+				}
+				if err := vs.dbc.PutAdvisoryDetail(tx, cveID, platformName, pkg.Name, advisory); err != nil {
+					return xerrors.Errorf("failed to save Alma advisory: %w", err)
+				}
+
+				vuln := types.VulnerabilityDetail{
+					Severity:         generalizeSeverity(erratum.Severity),
+					References:       references,
+					Title:            erratum.Title,
+					Description:      erratum.Description,
+					PublishedDate:    utils.MustTimeParse(time.UnixMilli(erratum.IssuedDate.Date).Format(time.RFC3339)),
+					LastModifiedDate: utils.MustTimeParse(time.UnixMilli(erratum.UpdatedDate.Date).Format(time.RFC3339)),
```

Comment:
These fields are for NVD.
https://github.com/aquasecurity/trivy-db/blob/62aa4616130e5e27b1eb33bf65a2e6b44090fc43/pkg/vulnsrc/vulnerability/vulnerability.go#L58

Could you also add comments in the following fields? `// Take from NVD` or something like that.
https://github.com/aquasecurity/trivy-db/blob/485692c1f0cb97b401e970bb78b16ac7834cd1c0/pkg/types/types.go#L96-L97
https://github.com/aquasecurity/trivy-db/blob/485692c1f0cb97b401e970bb78b16ac7834cd1c0/pkg/types/types.go#L142-L143